lumicks/pylake/correlated_stack.py

```diff
@@ -340,6 +340,59 @@ def select_frame(event):
 
         fig.canvas.mpl_connect('button_press_event', select_frame)
 
+    def export_tiff(self, file_name, roi=None):
+        """Export a video of a particular scan plot
+
+        Parameters
+        ----------
+        file_name : str
+            File name to export to.
+        roi : list_like
+            region of interest in pixel values [xmin, xmax, ymin, ymax]
+        """
+        from . import __version__ as version
+
+        def parse_tags(frame):
+            # Parse original file tags into list of tuples [(code, dtype, count, value, writeonce)]
+            # Only tags that are not resolved automatically by `TiffWriter.save()` are needed
+            n_channels = 3 if frame.is_rgb else 1
+            datetime = frame._src.tags["DateTime"].value
+            output_tags = ((274, "H", 1, 1),                            # Orientation
+                           (339, "H", n_channels, (1, )*n_channels),    # SampleFormat
+                           (306, "s", len(datetime), datetime))         # DateTime
+            return output_tags
+
+        # get ROI coordinates
+        if roi is not None:
+            xmin, xmax, ymin, ymax = roi
+        else:
+            ymin, xmin = 0, 0
+            ymax = self._get_frame(0).data.shape[0]
+            xmax = self._get_frame(0).data.shape[1]
+
+        # re-name alignment matrices fields in image description
+        # to reflect the fact that the image has already been processed
+        description = self._get_frame(0)._src.description
+        if description != "":
+            description = json.loads(description)
+            for color in ("blue", "green", "red"):
+                key = f"Alignment {color} channel"
+                description["Applied "+key] = description.pop(key)
+
+        # add pylake to Software tag
+        software = self._get_frame(0)._src.tags["Software"].value
+        software += f", pylake v{version}"
+
+        # write frames sequentially
+        with tifffile.TiffWriter(file_name) as tif:
+            for frame in self:
+                tif.save(frame.data[ymin:ymax, xmin:xmax], 
+                         description=json.dumps(description, indent=4),
```

Comment:
`tifffile` only saves the description on the first page. I thought it was cleaner to just keep it in the loop instead of having a separate conditional statement to check for the first page

Reply:
I'm not very knowledgeable about python but my impression would be that this code would call `json.dumps(...)` every iteration of the loop, and send the result to the function.

My suggestion would be to avoid this doing this (no conditional needed):
```python
            description = json.dumps(description, indent=4)
            for frame in self:
                tif.save(frame.data[ymin:ymax, xmin:xmax],
                         description=description,
```

Reply:
I'm not very knowledgeable about python but my impression would be that this code would call `json.dumps(...)` every iteration of the loop, and send the result to the function.

My suggestion would be to avoid this doing this (no conditional needed):
```python
            description = json.dumps(description, indent=4)
            for frame in self:
                tif.save(frame.data[ymin:ymax, xmin:xmax],
                         description=description,
```

Reply:
oh, I thought you were talking about something else ... good catch